src/AutorestSwift/Models/Complex/Schema.swift

```diff
@@ -100,15 +101,31 @@ class Schema: Codable, LanguageShortcut {
                 fatalError("Type mismatch. Expected number type but got \(self)")
             }
             swiftType = numberSchema.swiftType()
-        case .dateTime,
-             .date,
-             .time,
-             .unixTime:
-            swiftType = "Date"
+        case .dateTime:
+            if let dateSchema = self as? DateTimeSchema {
+                switch dateSchema.format {
+                case .dateTime:
+                    swiftType = "Iso8601Date"
+                case .dateTimeRfc1123:
+                    swiftType = "Rfc1123Date"
+                }
+            } else {
+                fatalError("Type .dateTime but not schema `DateTimeSchema`")
+            }
+        case .date:
+            swiftType = "SimpleDate"
```

Comment:
The name 'SimpleDate' seem confusing when there are `Iso88601Date', 'Rfc1123Date' which shows what standard they implement.

Reply:
True. I just didn't know a good format name for YMD. I also considered whether to make all these date types actually return their value (and thus type Date) and simply use these internally.